File: HW1/src/flower.cpp
//------------------------------------------------------------------------------
// flower.cpp - содержит функции обработки цветка
//------------------------------------------------------------------------------

#include "flower.h"

//------------------------------------------------------------------------------
// Ввод параметров цветка из потока
void In(flower &f, std::ifstream &ifst) {
    int place;
    ifst >> place >> f.name;
    switch (place) {
        case 1:
            f.flow_type = flower::flower_type::DOMESTIC;
            break;
        case 2:
            f.flow_type = flower::flower_type::GARDEN;
            break;
        case 3:
            f.flow_type = flower::flower_type::WILD;
            break;
        default:
            f.flow_type = flower::flower_type::INVALID_PLACE_NUMBER;
    }
}

// Случайный ввод параметров цветка
void InRnd(flower &f) {
    switch (Random() % 3) {
        case 0:
            f.flow_type = flower::flower_type::DOMESTIC;
            break;
        case 1:
            f.flow_type = flower::flower_type::GARDEN;
            break;
        case 2:
            f.flow_type = flower::flower_type::WILD;
            break;
        default:
            f.flow_type = flower::flower_type::INVALID_PLACE_NUMBER;
    }
    for (size_t i = 0; i < MAX_NAME_SIZE; ++i) {
        f.name[i] = static_cast<char>('a' + Random());
    }
    f.name[MAX_NAME_SIZE - 1] = '\0';
}

//------------------------------------------------------------------------------
// Вывод параметров цветка в поток
void Out(flower &f, std::ofstream &ofst) {
    ofst << "It is flower: flower type = ";
    switch (f.flow_type) {
        case flower::DOMESTIC:
            ofst << "domestic";
            break;
        case flower::GARDEN:
            ofst << "garden";
            break;
        case flower::WILD:
            ofst << "wild";
            break;
        case flower::INVALID_PLACE_NUMBER:
            ofst << "Invalid place number!";
    }
    ofst << ", name = " << f.name
         << ". Quotient = " << Quotient(f) << "\n";
}

//------------------------------------------------------------------------------
// Вычисление частного от деления числа гласных букв
// в названии цветка на общую длину его названия

double Quotient(flower &f) {
    // переменная для подсчёта гласных в названии цветка
    size_t number_of_vowels = 0;
    for (auto it: f.name) {
        char current_symbol = tolower(it);
        if (current_symbol == 'a' || current_symbol == 'e' || current_symbol == 'i' ||
            current_symbol == 'o' || current_symbol == 'u' || current_symbol == 'y') {
            ++number_of_vowels;
        }
    }
    return number_of_vowels / (double) strlen(f.name);
}

File: HW3/src/flower.py
from enum import Enum
import myRandom

#----------------------------------------------
# Перечисляемый тип представляющий типа цветка
class FlowerType(Enum):
    DOMESTIC = 1
    GARDEN = 2
    WILD = 3
    INVALID_PLACE_NUMBER = 0

# Класс цветка
class Flower :
    # Конструктор цветка
    def __init__(self):
        self.name = ""
        self.flowerType = FlowerType.INVALID_PLACE_NUMBER.name
    
    # Считывание параметров цветка из строки
    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        self.name = strArray[i]
        self.flowerType = FlowerType(int(strArray[i + 1]) 
            if int(strArray[i + 1]) <= 3 and int(strArray[i + 1]) >= 0 else 0).name
        i += 2
        return i

    # Случайная генерация параметров цветка
    def RandomIn(self):
        self.name = myRandom.randName()
        self.flowerType = FlowerType(myRandom.randInt(1, 3)).name

    # Вывод данных цветка в файл
    def Write(self, ostream):
        ostream.write("Flower: name = {}, flower type = {}, quotient = {}".format(self.name, self.flowerType, self.Quotient()))
        pass

    # Вычисление частного от деления числа гласных букв
    # в названии цветка на общую длину его названия
    def Quotient(self):
        numberOfVowels = 0
        for ch in self.name:
            if(ch == 'a' or ch == 'e' or ch == 'i' or ch == 'o' or ch == 'u' or ch == 'A'
                or ch == 'E' or ch == 'I' or ch == 'O' or ch == 'U'):
                numberOfVowels += 1
        return numberOfVowels / len(self.name)
